indico/modules/events/registration/controllers/management/reglists.py

```diff
@@ -277,18 +277,23 @@ def _get_user_data(self, user):
         user_data['title'] = get_title_uuid(self.regform, user_data['title'])
         return user_data
 
-    def _process(self):
-        form = make_registration_form(self.regform, management=True)()
-        if form.validate_on_submit():
-            data = form.data
-            session['registration_notify_user_default'] = notify_user = data.pop('notify_user', False)
-            create_registration(self.regform, data, management=True, notify_user=notify_user)
+    def _process_POST(self):
+        schema = make_registration_schema(self.regform)()
+        errors = schema.validate(request.json)
+
+        if not errors:
+            form = schema.load(request.json)
+            session['registration_notify_user_default'] = notify_user = form.pop('notify_user', False)
+            create_registration(self.regform, form, management=True, notify_user=notify_user)
             flash(_('The registration was created.'), 'success')
             return redirect(url_for('.manage_reglist', self.regform))
-        elif form.is_submitted():
+        else:
             # not very pretty but usually this never happens thanks to client-side validation
-            for error in form.error_list:
-                flash(error, 'error')
+            for field in errors:
+                flash(f'{field}: {errors[field]}', 'error')
```

Comment:
This can go away, submitting a regform should never redirect or return HTML; we can instead nicely show any errors on the client side by simply returning them in the usual webargs error format (so you may want to make sure to actually use the webargs flaskparser to process the form data)